src/controllers/connectionManager.ts

```diff
@@ -1115,6 +1129,16 @@ export default class ConnectionManager {
                 } else {
                     return false;
                 }
+            } else if (connection.errorNumber === Constants.errorSSLCertificateValidationFailed) {
+                const updatedConnection = await this.handleSSLError(
+                    fileUri,
+                    connectionCreds as IConnectionProfile,
+                );
+                if (updatedConnection) {
+                    await this.connect(fileUri, updatedConnection);
```

Comment:
`return await this.connect()`, right?  And likely the same on line 1128 above, I think.